core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -129,25 +131,30 @@ private LocalDate getCurrentDayDate(UserAnonymizedDTO userAnonymized)
 				activityEntitiesByDate.put(date, new HashSet<T>());
 			}
 			Set<T> activityEntitiesAtDate = activityEntitiesByDate.get(date);
-			activeGoals.forEach(g -> addMissingInactivity(g, dateAtStartOfDay, activityEntitiesAtDate, userAnonymized,
+			activeGoals.forEach(g -> addMissingInactivity(g, dateAtStartOfInterval, activityEntitiesAtDate, userAnonymized,
 					inactivityEntitySupplier));
 		}
 	}
 
-	private Set<Goal> getActiveGoals(UserAnonymizedDTO userAnonymized, ZonedDateTime dateAtStartOfDay, ChronoUnit timeUnit)
+	private Set<Goal> getActiveGoals(UserAnonymizedDTO userAnonymized, ZonedDateTime dateAtStartOfInterval, ChronoUnit timeUnit)
 	{
 		Set<Goal> activeGoals = userAnonymized.getGoals().stream()
-				.filter(g -> g.getCreationTime().isBefore(dateAtStartOfDay.plus(1, timeUnit))).collect(Collectors.toSet());
+				.filter(g -> wasGoalActiveAtInterval(g, dateAtStartOfInterval, timeUnit)).collect(Collectors.toSet());
 		return activeGoals;
 	}
 
-	private <T extends IntervalActivity> void addMissingInactivity(Goal activeGoal, ZonedDateTime dateAtStartOfDay,
+	private boolean wasGoalActiveAtInterval(Goal goal, ZonedDateTime dateAtStartOfInterval, ChronoUnit timeUnit)
```

Comment:
Can you make clear from the code why you are adding one `timeunit` to the give date?